### ISSUE TITLE
ViewControllers are tied together

### ISSUE THREAD

**Issue by ngzhian** [midichrolians/midichrolian-ios#14]
For example, our main `ViewController` mucks around too much with `SampleTableViewController`, `AnimationTableViewController`. In the original code, we are confusing a `ViewController` with a `Controller` in MVC.
I'll start a series a small PRs to slowly tease apart these pieces.
What I hope to achieve eventually is for distinct parts of the app to be composed of `Controller` that manages `ViewController`, which is kind of like the `ViewModel` layer in MVVM, and `View`.

TLDR; coupling no good, layer and abstraction good, trying to be good

![class like diagram](https://cloud.githubusercontent.com/assets/1749303/24233961/69333952-0fd0-11e7-92cd-cef6d97e2fd4.png)

- [x] #15 
- [x] Extract top nav controller #17
- [x] Extract grid collection into a grid controller #24 
- [x] Chain delegations correctly #18
- [x] think of how to define view constraints when we don't have direct access to the views
